File: services/project/internal/entity/project.go
package entity

import (
	"time"

	"github.com/google/uuid"
)

// message Project {
//     string id = 1;
//     string title = 2;
//     string description = 3;
//     string status = 4;
//     string organization_id = 5;

//     google.protobuf.Timestamp created_at = 6;
//     google.protobuf.Timestamp updated_at = 7;
// }

type Project struct {
	ID          uuid.UUID `db:"id"`
	Title       string    `db:"title"`
	Description string    `db:"description"`
	Status      string    `db:"status"`
	CompanyID   string    `db:"company_id"`
	CreatedAt   time.Time `db:"created_at"`
	UpdatedAt   time.Time `db:"updated_at"`
}

File: services/frontend/src/lib/project/project.ts
import { Pagination } from "../utils/list";

export type Project = {
  id: string;
  title: string;
  code_name: string;
  description: string;
  status: string;
  company_id: string;
  created_at: {
    seconds: number;
  };
  updated_at: {
    seconds: number;
  };
  total_tasks: number;
  done_tasks: number;
};

export type ProjectCreation = Pick<
  Project,
  "title" | "description" | "company_id" | "code_name"
>;

export type ProjectFilters = Pagination & {
  company_id?: string;
  title?: string;
  status?: string;
};
